src/GraphQL.Tests/Types/AutoRegisteringInterfaceGraphTypeTests.cs

```diff
@@ -0,0 +1,809 @@
+        graphType.Fields.Count.ShouldBe(2);
+    }
+
+    [Fact]
+    public async Task CustomHardcodedArgumentAttributesWork()
+    {
+        var graphType = new AutoRegisteringInterfaceGraphType<CustomHardcodedArgumentAttributeTestInterface>();
+        var fieldType = graphType.Fields.Find(nameof(CustomHardcodedArgumentAttributeTestInterface.FieldWithHardcodedValue))!;
+        var resolver = fieldType.Resolver!;
+        (await resolver.ResolveAsync(new ResolveFieldContext
+        {
+            Source = new CustomHardcodedArgumentAttributeTestClass(),
+        }).ConfigureAwait(false)).ShouldBe("85");
+    }
+
+    [Fact]
+    public void TestInheritedFields()
+    {
+        var graphType = new AutoRegisteringInterfaceGraphType<InheritanceTests>();
+        graphType.Fields.Find("Field1").ShouldNotBeNull();
+        graphType.Fields.Find("Field2").ShouldNotBeNull();
+        graphType.Fields.Find("Field3").ShouldNotBeNull();
+        graphType.Fields.Find("Field4").ShouldNotBeNull();
+        graphType.Fields.Find("Field5").ShouldNotBeNull();
+        graphType.Fields.Count.ShouldBe(5);
+    }
+
+    [Theory]
+    [InlineData("{find(type:CAT){id name}}", @"{""data"":{""find"":{""id"":""10"",""name"":""Fluffy""}}}")]
+    [InlineData("{find(type:DOG){id name}}", @"{""data"":{""find"":{""id"":""20"",""name"":""Shadow""}}}")]
+    [InlineData("{find(type:CAT){id name ... on Cat { lives } ... on Dog { isLarge }}}", @"{""data"":{""find"":{""id"":""10"",""name"":""Fluffy"",""lives"":9}}}")]
+    [InlineData("{find(type:DOG){id name ... on Cat { lives } ... on Dog { isLarge }}}", @"{""data"":{""find"":{""id"":""20"",""name"":""Shadow"",""isLarge"":true}}}")]
+    [InlineData("{cat{id name}}", @"{""data"":{""cat"":{""id"":""10"",""name"":""Fluffy""}}}")]
+    [InlineData("{dog{id name}}", @"{""data"":{""dog"":{""id"":""20"",""name"":""Shadow""}}}")]
+    [InlineData("{cat{...frag}} fragment frag on IAnimal { id name }", @"{""data"":{""cat"":{""id"":""10"",""name"":""Fluffy""}}}")]
+    [InlineData("{dog{...frag}} fragment frag on IAnimal { id name }", @"{""data"":{""dog"":{""id"":""20"",""name"":""Shadow""}}}")]
+    public async Task EndToEndTest(string query, string expected)
+    {
+        var services = new ServiceCollection();
+        services.AddGraphQL(b => b
+            .AddAutoSchema<TestQuery>()
+            .AddSystemTextJson());
+        using var provider = services.BuildServiceProvider();
+        var executer = provider.GetRequiredService<IDocumentExecuter>();
+        var result = await executer.ExecuteAsync(new()
+        {
+            Query = query,
+            RequestServices = provider,
+        }).ConfigureAwait(false);
+        var serializer = provider.GetRequiredService<IGraphQLTextSerializer>();
+        var actual = serializer.Serialize(result);
+        actual.ShouldBeCrossPlatJson(expected);
+    }
+
+    [Fact]
+    public async Task ExecutesQueryWithInterfaceOnly()
```

Comment:
Verifies that an object can be returned from an interface even if it does not match any object graph types registered to the interface.  (Failed previously.)  With this change any graph can return an GraphQL interface even if the schema does not contain any object types that implement that GraphQL interface.